core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1398,6 +1398,21 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     }
   }
 
+  case object RegisterBrokerAndReelect extends ControllerEvent {
```

Comment:
We have the following code in KafkaController. We just need to replace Reelect with RegisterBrokerAndReelect in afterInitializingSession().

```
    zkClient.registerStateChangeHandler(new StateChangeHandler {
      override val name: String = StateChangeHandlers.ControllerHandler
      override def onReconnectionTimeout(): Unit = error("Reconnection timeout.")
      override def afterInitializingSession(): Unit = {
        eventManager.put(Reelect)
      }
```